### ISSUE TITLE
Add support for AWS Graviton

### ISSUE THREAD

**Issue by zamazan4ik** [tarantool/tarantool#5471]
**Tarantool version**: whatever

**OS version**: Linux-based

**Bug description**: as far as I see the only officially supported hardware architecture for now is x86-64. We have a case when we want to build cloud-native application, which will be deployed to AWS EC2 instances with AWS [Graviton](https://aws.amazon.com/ec2/graviton/) CPUs (just because they are cheaper for the same performance in our workload).

So will be nice to see official support for AWS Graviton processors in Tarantool.

**Steps to reproduce**: install Tarantool on AWS Graviton EC2 instance and run it :)

**Comment by urandon** [comment]
Oh! We have the same problem with Tarantool! Will be nice to get such feature ASAP

**Comment by tsafin** [comment]
**TL;DR** - it works with 1 simple patch

**The longer version below:**

If you create t4g.micro instance running Amazon Linux and then install all the required RPM packages, namely:

```
yum install -y \
    git perl gcc cmake make gcc-c++ libstdc++-static autoconf automake libtool \
    python-msgpack python-yaml python-argparse python-six python-gevent \
    openssl-devel readline-devel libicu-devel
```

then it's able to complete cmake configuration command
```
mkdir build && cd build
cmake .. -DCMAKE_BUILD_TYPE=RelWithDebInfo  -DCMAKE_INSTALL_PREFIX=$HOME/share -DENABLE_DIST=ON
```

But please be careful with parallel `make` because of all constraint of created t4g instance! The safer approach is to use -j1
```
make -j1 install
```
_There was small problem though_ - fiber.top was not properly disabled as it's implemented currently only for x86/x64 targets, i.e. this simple fix was necessary for current arm64 host:

```patch
diff --git a/src/lib/core/fiber.h b/src/lib/core/fiber.h
index 539e5c8e7..70c5cbf2f 100644
--- a/src/lib/core/fiber.h
+++ b/src/lib/core/fiber.h
@@ -49,7 +49,7 @@
  * Fiber top doesn't work on ARM processors at the moment,
  * because we haven't chosen an alternative to rdtsc.
  */
-#ifdef __CC_ARM
+#ifdef __aarch64__
 #define ENABLE_FIBER_TOP 0
 #else
 #define ENABLE_FIBER_TOP 1
```

So finally tarantool is up'n'running, with apparently functioning LuaJIT and FFI:

```
[ec2-user@ip-172-3x-4x-2xx build]$ ./src/tarantool
Tarantool 2.7.0-20-g99d6c8a40
type 'help' for interactive help

tarantool> _VERSION
---
- Lua 5.1
...

tarantool> _TARANTOOL
---
- 2.7.0-20-g99d6c8a40
...

tarantool> package.loaded.tarantool
---
- version: 2.7.0-20-g99d6c8a40
  build:
    target: Linux-aarch64-RelWithDebInfo
    options: cmake . -DCMAKE_INSTALL_PREFIX=/home/ec2-user/share -DENABLE_BACKTRACE=OFF
    mod_format: so
    flags: ' -fexceptions -funwind-tables -fno-common -fopenmp -std=c11 -Wall -Wextra
      -Wno-strict-aliasing -Wno-char-subscripts -Wno-format-truncation -Wno-gnu-alignof-expression
      -fno-gnu89-inline -Wno-cast-function-type'
    compiler: /usr/bin/cc /usr/bin/c++
  pid: 'function: 0xf871091b00'
  package: Tarantool
  uptime: 'function: 0xf871091ac8'
...

tarantool> jit = require 'jit'
---
...

tarantool> jit.arch
---
- arm64
...

```
```

**Comment by zamazan4ik** [comment]
@kyukhin since you've closed the issue - what is the current status of Tarantool on AWS Graviton and AWS Graviton2? Are they officially supported?

**Comment by zamazan4ik** [comment]
By the way, Graviton3 CPUs are in preview now and you can try to test Tarantool on it before the Graviton3 GA: https://aws.amazon.com/ru/blogs/aws/join-the-preview-amazon-ec2-c7g-instances-powered-by-new-aws-graviton3-processors/

**Comment by Totktonada** [comment]
We land some aarch64 related fixes and now we have Graviton machines in CI (@kyukhin may reveal details — I don't know exact configuration). I would not say that it is bug free (at least our LuaJIT fork requires downstreaming of [some](https://github.com/tarantool/tarantool/wiki/Vanilla-LuaJIT-sync-status) GC64 related bugfixes), but it passes our test suite. I would call it 'preliminary support'.

@igormunkin May reveal our plans around aarch64.

Everything said is applicable only for tarantool 2.10.0-beta1 and newer.
